xonsh/prompt/vc.py

```diff
@@ -175,19 +164,31 @@ def current_branch():
 
 
 def _git_dirty_working_directory(q, include_untracked):
-    status = None
     denv = builtins.__xonsh__.env.detype()
     try:
-        cmd = ["git", "status", "--porcelain"]
+        # Borrowed from this conversation
+        # https://github.com/sindresorhus/pure/issues/115
         if include_untracked:
-            cmd.append("--untracked-files=normal")
+            cmd = [
+                "git",
+                "status",
+                "--porcelain",
+                "--quiet",
+                "--untracked-files=normal",
+            ]
         else:
-            cmd.append("--untracked-files=no")
-        status = subprocess.check_output(cmd, stderr=subprocess.DEVNULL, env=denv)
+            unindexed = ["git", "diff", "--no-ext-diff", "--quiet"]
+            indexed = unindexed + ["--cached", "HEAD"]
```

Comment:
This gets broken (also on current `master`) if there exists a file called `HEAD` in the repo.

Solution: Append another arg `--`: https://stackoverflow.com/questions/2657935/checking-for-a-dirty-index-or-untracked-files-with-git/2659808#comment76137052_2659808